learn/why-ballerina/flexibly-typed.md

```diff
@@ -175,11 +182,11 @@ type Person record {|
 |};
 ```
 
-Here, the type `Person` is an open record type, the notation `json...;` denotes that this record type can contain additional fields that are not mentioned in the record type descriptor, as long as these additional fields belong to type `json`.
+Here, the type `Person` is an open record type, the notation `json...;` denotes that this record type can contain additional fields that are not explicitly mentioned in the record type descriptor, as long as these additional fields belong to type `json`.
 
-The earlier `DoorState` record type was defined explicitly as a closed record type. Therefore, you were able to list out all the possible shapes in the `DoorState` type. If this type was defined as an open record, you would have an infinite number of shapes since `DoorState` values can have any arbitrary fieldset in the code.
+The earlier `DoorState` record type was defined explicitly as a closed record type. Therefore, you were able to list out all the possible shapes in the `DoorState` type. If this type was defined as an open record, you would have an infinite number of shapes since `DoorState` values can have any arbitrary set of fields in the code.
 
-The `Person` record type above has an <a href="/learn/by-example/optional-fields">optional field</a> `creditScore` (denoted by the suffix `"?"`). This means the field value of `creditScore` of a `Person` record can be skipped without setting a value. Later on, this field can be accessed using the `"?."` operator, which would return a value of type `CreditScore?`, which is equivalent to the union type `CreditScore|()`. In Ballerina, the nil value and the type are represented by `()`.
+The `Person` record type above has an <a href="/learn/by-example/optional-fields">optional field</a> `creditScore` (denoted by the suffix `"?"`). This means the `creditScore` field may not be set (i.e., not mandatory) when creating a value of type `Person`. Later on, this field can be accessed using field access (`"."`) or optional field access (`"?."`). The static type of this access would be `CreditScore?` which is equivalent to the union type `CreditScore|()`. At runtime, accessing the optional field will return the value (belonging to the type of the field `CreditScore`) if the field is present, else if the field is not present it will return nil. In Ballerina, the nil value and the type are represented by `()`. Field access can be used to access optional fields only when the type of the field does not contain nil (such as `creditScore` in this example).
```

Comment:
```suggestion
The `Person` record type above has an <a href="/learn/by-example/optional-fields">optional field</a> `creditScore` (denoted by the suffix `"?"`). This means the `creditScore` field may not be set (i.e., not mandatory) when creating a value of type `Person`. Later on, this field can be accessed using field access (`"."`) or optional field access (`"?."`). The static type of this access would be `CreditScore?`, which is equivalent to the union type `CreditScore|()`. At runtime, accessing the optional field will return the value (belonging to the type of the field `CreditScore`) if the field is present, else, if the field is not present it will return nil. In Ballerina, the nil value and the type are represented by `()`. Field access can be used to access optional fields only when the type of the field does not contain nil (such as `creditScore` in this example).
```